core/src/main/java/io/grpc/CallOptions.java

```diff
@@ -141,8 +141,8 @@ public CallOptions withRequestKey(@Nullable RequestKey requestKey) {
    */
   @ExperimentalApi
   @Nullable
-  public RequestKey getRequestKey() {
-    return requestKey;
+  public Attributes getAttributes() {
```

Comment:
If we treat it as a generic attributes container that is not specific to load-balancing affinity, I would expect to be able to update an individual key:
```
callOptions = callOptions.withAttribute(key, value);
```
and to have a similar API on `AbstractStub`.

We don't seem to gain much from putting affinity key and other potential call attributes in the same container. I would prefer name the field as `affinityData`, then it makes sense to update it as a whole.

@ejona86 What's your opinion?